### PR TITLE
Feature/permissions conversation options

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

In the group details screen, if the self user is a `collaborator`, then they should not see any UI that allows them to:

1. Rename the conversation
2. Change the group access (guests)
3. Change the timeout
4. Change the read receipts

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/GroupDetails/Cells/GroupDetailsRenameCell.swift [hunk 1]
```diff
@@ -80,6 +84,11 @@ class GroupDetailsRenameCell : UICollectionViewCell {
         contentStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16).isActive = true
         contentStackView.spacing = 8
         
+        if !selfUserIsAuthorized {
+            titleTextField.isUserInteractionEnabled = false
+            accessoryIconView.isHidden = true
+        }
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Cells/GroupDetailsRenameCell.swift]: We can move these check to func configure(for conversation: ZMConversation), so the UI will response to the role change event.

**Comment by billypchan** [comment]
@johnxnguyen we can merge this PR after the conflicts are fixed.

File: Wire-iOS Tests/GroupDetailsViewControllerSnapshotTests.swift [hunk 2]
```diff
@@ -22,45 +22,84 @@ import XCTest
 final class GroupDetailsViewControllerSnapshotTests: CoreDataSnapshotTestCase {
     
     var sut: GroupDetailsViewController!
+    var groupConversation: ZMConversation!
```
Comment (billypchan) [inline Wire-iOS Tests/GroupDetailsViewControllerSnapshotTests.swift]: This is a nice a helper and we can move it to CoreDataSnapshotTestCase later.

Comment (johnxnguyen) [reply Wire-iOS Tests/GroupDetailsViewControllerSnapshotTests.swift]: `CoreDataSnapshotTestCase` already has the `createGroupConversation()` helper, but I needed to configure my own group.

**Pull request merged**